js/pokemon.js

```diff
@@ -357,17 +370,49 @@ function revealPokemon(correctlyGuessed, language) {
 
 function generateNewNumbers(force) {
 
-    if(force || (currentGen !== newGen)) {
-        var i=0, j;
-        
+    if(force || !sortCompareArrays(currentGen, newGen)) {
+
+        console.log("generating New Pokemon array")
+
         upcomingPokemon = new Array();
         upcomingPokemonArrayPos = 0;
-        
-        for(j=minPokemonNumber; j<=maxPokemonNumber; j++) {
-            upcomingPokemon[i] = j;
-            i++;
+        var i = 0;
```

Comment:
Where is this i being used? It looks like it will always be the same as j in the loop below and not used elsewhere